srcjs/shinyapp.js

```diff
@@ -714,6 +714,131 @@ var ShinyApp = function() {
     });
   });
 
+  addMessageHandler('shiny-insert-tab', function(message) {
+    var $tabsetPanel = $("#" + message.tabsetPanelId);
+    if ($tabsetPanel.length === 0) {
+      throw 'There is no tabsetPanel with id ' + message.tabsetPanelId;
+    };
+
+    // This is the JS equivalent of the builtItem() R function that is used
```

Comment:
I'm not so familiar with `buildTabset` but I assume the answer to Winston's other question about `navbarPage`, `navbarMenu`, and `navlistPanel` would be a factor?